pymongo/collection.py

```diff
@@ -2250,7 +2248,7 @@ def list_indexes(
         self,
         session: Optional[ClientSession] = None,
         comment: Optional[Any] = None,
-    ) -> CommandCursor[MutableMapping[str, Any]]:
+    ) -> CommandCursor[_DocumentType]:
```

Comment:
I believe list_indexes always returns MutableMapping. Could you confirm?

_DocumentType is intended to represent documents in the collection, not index documents, so let's avoid using it here.